internal/ir/infra.go

```diff
@@ -0,0 +1,65 @@
+package ir
+
+import (
+	cfgv1a1 "github.com/envoyproxy/gateway/api/config/v1alpha1"
+)
+
+// Infra defines managed infrastructure.
+type Infra struct {
+	// Proxy defines managed proxy infrastructure.
+	Proxy *ProxyInfra
```

Comment:
since the infra manager layer is platform specific, it makes sense to generalize this structure, either as an interface sharing the same methods or splitting up via fields. This is internal and can be changed in the near future, so should be fine for now.